Resolver/Resolver.cs

```diff
@@ -245,15 +267,34 @@ static AddressResult[] ParseResponse(Span<byte> buffer, ref Header header)
             }
         }
 
+        public ValueTask<TxtResult[]> ResolveTextAsync(string name, CancellationToken cancellationToken = default)
+        {
+            return SendQueryAsync(name, QueryType.Text, ParseResponse, cancellationToken);
+
+            static TxtResult[] ParseResponse(Span<byte> buffer, ref Header header)
+            {
+                int offset = SkipResponseQuestionSection(buffer, header.QueryCount);
+                var result = new TxtResult[header.AnswerCount];
+                int actualCount = ParseTxtRecords(buffer, ref offset, result);
+
+                if (actualCount != result.Length)
+                {
+                    throw new Exception($"Invalid response: expected {result.Length} TXT records, got {actualCount}.");
+                }
+
+                return result;
+            }
+        }
+
         public async ValueTask<ServiceResult[]> ResolveServiceAsync(string name, CancellationToken cancellationToken = default)
-            => (await ResolveServiceAsync(name, false, cancellationToken)).Item1;
+            => (await ResolveServiceAsync(name, false, cancellationToken)).Services;
 
         // https://www.rfc-editor.org/rfc/rfc2782.html: "Implementors are urged, but not required, to return the address record(s) in the Additional Data section."
         // If no matching addresses are found, an empty array is being returned.
         public async ValueTask<(ServiceResult[] Services, AddressResult[] Addresses)> ResolveServiceAndAddressesAsync(string name, CancellationToken cancellationToken = default)
         {
             var result = await ResolveServiceAsync(name, true, cancellationToken);
-            Debug.Assert(result.Addresses != null);
+            Assert(result.Addresses != null);
```

Comment:
would this crash the process?

Reply:
It throws instead. I ran into an assertion failure in a test, so I turned all of them into exceptions temporarily. Easier to debug for me.